torch/_dynamo/symbolic_convert.py

```diff
@@ -3198,25 +3217,28 @@ def get_trace_call_log_str():
         except Exception as e:
             log.debug("FAILED INLINING %s", code)
             raise
-        assert tracer.symbolic_result is not None
+        assert self.symbolic_result is not None
 
-        if tracer.f_globals is parent.f_globals:
+        if self.f_globals is parent.f_globals:
             # Merge symbolic_globals back if parent and child are in the same namespace
-            parent.symbolic_globals.update(tracer.symbolic_globals)
+            parent.symbolic_globals.update(self.symbolic_globals)
 
-        parent.inconsistent_side_effects |= tracer.inconsistent_side_effects
+        parent.inconsistent_side_effects |= self.inconsistent_side_effects
 
         log.debug("DONE INLINING %s", code)
 
         if is_generator(code):
-            assert isinstance(tracer, InliningGeneratorInstructionTranslator)
-            assert tracer.symbolic_result.as_python_constant() is None
-            return ListIteratorVariable(
-                tracer.generated_items,
-                mutation_type=ValueMutationNew(),
-            )
+            assert isinstance(self, InliningGeneratorInstructionTranslator)
+            if not self.consume_all_items and self.generator_exhausted:
```

Comment:
Can you add more comment here? I don't fully understand whats going on here

The scenario I am thinking is - Suppose you have a generator and then you are calling next on it, and suppose you are met with a graph break. You will not be able to exhaust the generator in this case. But, since we dont know how to reconstruct a generator-that-has-been-invoked, we should graph break all the way to the parent frame. Is this happening already?

Reply:
> Is this happening already?

Yes! We graph break on the parent frame and the `__enter__/__exit__` frames are executed in eager. There are two tests (test_graph_break_inside_ctx_1/2) that asserts we are correctly handling them.

```python
# test_graph_break_inside_ctx_2

@contextlib.contextmanager
def whoo(x):
    try:
        torch._dynamo.graph_break()
        yield x.cos()
    finally:
        pass

def g(x):
    return x.neg() + x.acos()

def f(x):
    y = x.sin()
    with whoo(x) as z:
        y += g(z)
    y += y.tan()
    return y
```

In the case above, dynamo will produce a graph only for the frame `g`.